test/mocks/http/mocks.h

```diff
@@ -388,6 +388,14 @@ class MockAsyncClientRequest : public AsyncClient::Request {
   MockAsyncClient* client_;
 };
 
+class MockAsyncClientStream : public AsyncClient::Stream {
+public:
```

Comment:
Yeah, I don't think this will work with the templates unless we instantiate for every type that is used.